R/extend_scenario_trajectory.R

```diff
@@ -109,3 +110,154 @@ extend_scenario_trajectory <- function(data,
       .data$scenario_geography, .data$ald_sector, .data$technology, .data$year
     )
 }
+
+
+#' Extend the scenario pathways based on the fair share approach (now known as
+#' market share approach).  We first ensure that all scenarios are extended to
+#' the same point in time (start of the analysis plus forecast time frame.
+#' Then we take the first value of the production on company level as the start
+#' value and from the second year onward, we calculate the level of production
+#' for a company under the scenarios at hand by applying the relative changes
+#' for the respective years to the absolute values of the previous year.
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   (in the portfolio). Pre-processed to fit analysis parameters and after
+#'   conversion of power capacity to generation.
+#' @param scenario_data A data frame containing scenario data for the specified
+#'   parameters of the analysis, including the business as usual and target
+#'   scenarios, the relevant scenario geography and time frame for each of the
+#'   technologies.
+#' @param start_analysis Numeric. A vector of length 1 indicating the start
+#'   year of the analysis.
+#' @param end_analysis Numeric. A vector of length 1 indicating the end
+#'   year of the analysis.
+#' @param time_frame Numeric. A vector of length 1 indicating the number of years
+#'   for which forward looking production data is considered.
+#' @noRd
+extend_scenario_trajectory <- function(data,
+                                       scenario_data = NULL,
```

Comment:
I am conceptually to detached to understand/review this, sorry. If you are not comfortable merging this without approval let us synch on alternative strategy to increase confidence in the solution by tessting/running thorugh solution together